pkg/plugins/manager/loader/loader.go

```diff
@@ -185,7 +186,11 @@ func (l *Loader) loadPlugins(ctx context.Context, src plugins.PluginSource, foun
 		// Detect angular for external plugins
 		if p.IsExternalPlugin() {
 			var err error
-			p.AngularDetected, err = l.angularInspector.Inspect(p)
+
+			cctx, canc := context.WithTimeout(ctx, time.Minute*1)
```

Comment:
1 minute sounds high given that this happens when Grafana starts. Could it really take that long time to run the detection for one plugin?

Reply:
Agree, I was planning to take a better look at tweaking the timeout/refactor to background service similar to the signing keys retrieval after merging this initial PR. Currently it is under a feature flag, so it's safe to merge, but I can the refactoring into background service it directly in this PR as suggested by Andres. I believe once I do that, the context arg could go away (as the GCOM call will happen in the background) so it should solve this concern as well :smiley:

Reply:
Make sense :+1: